fe/fe-core/src/main/java/com/starrocks/load/loadv2/LoadingTaskPlanner.java

```diff
@@ -200,6 +215,87 @@ public void plan(TUniqueId loadId, List<List<TBrokerFileStatus>> fileStatusesLis
+        scanNode.setUseVectorizedLoad(true);
+        scanNode.init(analyzer);
+        scanNode.finalizeStats(analyzer);
+        LOG.info("use vectorized load: {}, load job id: {}", true, loadJobId);
+        scanNodes.add(scanNode);
+        descTable.computeMemLayout();
+
+        // 3. Scan plan fragment
+        PlanFragment scanFragment = new PlanFragment(new PlanFragmentId(0), scanNode, DataPartition.RANDOM);
+        scanFragment.setParallelExecNum(parallelInstanceNum);
+
+        fragments.add(scanFragment);
+
+        // 5. Exchange node
+        List<Column> columns = table.getFullSchema();
+
+        List<Column> keyColumns = table.getKeyColumnsByIndexId(table.getBaseIndexId());
+        List<Expr> partitionExprs = Lists.newArrayList();
+        keyColumns.forEach(column -> {
+            partitionExprs.add(new SlotRef(tupleDesc.getColumnSlot(column.getName())));
+        });
+
+        DataPartition dataPartition = new DataPartition(TPartitionType.HASH_PARTITIONED, partitionExprs); 
+        ExchangeNode exchangeNode = new ExchangeNode(new PlanNodeId(nextNodeId++), scanFragment.getPlanRoot(),
+                dataPartition);
+        PlanFragment sinkFragment = new PlanFragment(new PlanFragmentId(1), exchangeNode, dataPartition);
+        exchangeNode.setFragment(sinkFragment);
+        scanFragment.setDestination(exchangeNode);
+        scanFragment.setOutputPartition(dataPartition);
+
+        // 4. Olap table sink
+        List<Long> partitionIds = getAllPartitionIds();
+        OlapTableSink olapTableSink = new OlapTableSink(table, tupleDesc, partitionIds);
+        olapTableSink.init(loadId, txnId, dbId, timeoutS);
+        olapTableSink.complete();
+
+        // 6. Sink plan fragment
+        sinkFragment.setSink(olapTableSink);
+        // At present, we only support dop=1 for olap table sink.
+        // because tablet writing needs to know the number of senders in advance
+        // and guaranteed order of data writing
+        // It can be parallel only in some scenes, for easy use 1 dop now.
+        sinkFragment.setPipelineDop(1);
+        sinkFragment.setParallelExecNum(parallelInstanceNum);
+        // After data loading, we need to check the global dict for low cardinality string column
+        // whether update.
+        sinkFragment.setLoadGlobalDicts(globalDicts);
+
+        fragments.add(sinkFragment);
+
+        // 4. finalize
+        for (PlanFragment fragment : fragments) {
+            try {
+                fragment.finalize(analyzer, false);
+            } catch (NotImplementedException e) {
+                LOG.info("Fragment finalize failed.{}", e.getMessage());
+                throw new UserException("Fragment finalize failed.");
+            }
+        }
+        Collections.reverse(fragments);
+    }
+
+    public Boolean needShufflePlan() {
+        if (KeysType.DUP_KEYS.equals(table.getKeysType())) {
+            return false;
+        }
+
+        if (table.getDefaultReplicationNum() <= 1) {
```

Comment:
If the input file is one, we can use non-shuffle plan.

Reply:
seems not, since file can be split.